### PR TITLE
how-to: update doc for deploying tidb cluster using tiup

### PR DESCRIPTION
**Pull request by ran-huang**
### What is changed, added or deleted? (Required)

Update the doc for deploying TiDB cluster using TiUP.

### Which TiDB version(s) do your changes apply to? (Required)

- [x] master (the latest development version)
- [x] v4.0 (TiDB 4.0 versions)
- [ ] v3.1 (TiDB 3.1 versions)
- [ ] v3.0 (TiDB 3.0 versions)
- [ ] v2.1 (TiDB 2.1 versions)

**If you select two or more versions from above**, to trigger the bot to cherry-pick this PR to your desired release version branch(es), you **must** add corresponding labels such as **needs-cherry-pick-4.0**, **needs-cherry-pick-3.1**, **needs-cherry-pick-3.0**, and **needs-cherry-pick-2.1**.

### What is the related PR or file link(s)?

- This PR is translated from:https://github.com/pingcap/docs-cn/pull/3115, https://github.com/pingcap/docs-cn/pull/3330, https://github.com/pingcap/docs-cn/pull/3224, https://github.com/pingcap/docs-cn/pull/3416, https://github.com/pingcap/docs-cn/pull/3425, https://github.com/pingcap/docs-cn/pull/3417, https://github.com/pingcap/docs-cn/pull/3476/
- Other reference link(s): https://docs.pingcap.com/zh/tidb/v4.0/production-deployment-using-tiup

### CODE REVIEW

File: production-deployment-using-tiup.md [hunk 1]
```diff
@@ -170,1031 +74,39 @@ Log in to the control machine using a regular user account (take the `tidb` user
...
-The physical machines on which TiDB and TiKV components are deployed have a 2-way processor with 16 Vcores per way, and the memory also meets the standard.
-
-In order to improve the resource utilization, you can deploy multiple instances on a single machine, that is, you can bind the cores through numa to isolate CPU resources used by TiDB and TiKV instances.
+## Step 3: Edit the initialization configuration file
 
-You can also deploy PD and Prometheus in a mixed manner, but the data directories of the two require two independent file systems.
+According to the intended cluster topology, you need to manually create and edit the cluster initialization configuration file.
 
-#### Key parameter configuration
+The following examples cover six common scenarios. You need to create a YAML configuration file （named `topology.yaml` for example) according to the topology description and templates in the corresponding links. For other scenarios, edit the configuration accordingly.
```
Comment (ran-huang) [inline production-deployment-using-tiup.md:81]:
```suggestion
The following examples cover six common scenarios. You need to create a YAML configuration file (named `topology.yaml` for example) according to the topology description and templates in the corresponding links. For other scenarios, edit the configuration accordingly.
```

**Comment by yikeke** [comment]
Except for the deadlinks, rest LGTM. You can take a look at my commits and see if there are any problems. @ran-huang

**Comment by ran-huang** [comment]
@yikeke Made a minor edit. Rest LGTM.

**Pull request merged**